### ISSUE TITLE
Incorrectly parsed escaped strings?

### ISSUE THREAD

**Issue by mcseemk** [simd-lite/simd-json#307]
Looks like simd-json incorrectly parses escaped strings:

```rust
const STR: &str = r#"
["{\"arg\":\"test\"}"]
"#;

fn main() {
    let mut str_bytes = STR.to_string().into_bytes();
    let mut r: Vec<String> = simd_json::from_slice(&mut str_bytes).unwrap();
    let simd_string = r.pop().unwrap();
    let mut r: Vec<String> = serde_json::from_str(STR).unwrap();
    let serde_string = r.pop().unwrap();
    println!("simd-json decoded string: {}\nserde-json decoded string: {}", simd_string, serde_string);
    assert_eq!(simd_string, serde_string);
}
```

Output:

> simd-json decoded string: {\"arg\":\"tes
> serde-json decoded string: {"arg":"test"}
> thread 'main' panicked at src/main.rs:12:5:
> assertion `left == right` failed

Simd-json returned an incorrect string above.

**Comment by Licenser** [comment]
Hi I can't reproduce this with the current version of simd-json and AVX2, can you share some details on version & CPU architecture you're trying this on?

**Comment by mcseemk** [comment]
Hi, I figured out that it is not arch-related. 

If compiled in release mode on the last nightly (rustc 1.73.0-nightly (6ef7d16be 2023-08-19)) it fails everywhere I tried: Graviton1/2, AMD Epyc 7551P, AMD Ryzen 7950X. In debug mode works fine though.

On the last stable rustc 1.71.1 (eb26296b5 2023-08-03) it works fine in both modes.

**Comment by Licenser** [comment]
Odd,

[this](https://github.com/Licenser/simd-test) works for me on both nightly and test in release mode:

```
$ cargo +nightly run --release
   Compiling simd-test v0.1.0 (/Users/heinz/Projects/simd-test)
    Finished release [optimized] target(s) in 2.17s
     Running `target/release/simd-test`
simd-json decoded string: {"arg":"test"}
serde-json decoded string: {"arg":"test"}
```

```
 $ cargo run --release
    Finished release [optimized] target(s) in 0.22s
     Running `target/release/simd-test`
simd-json decoded string: {"arg":"test"}
serde-json decoded string: {"arg":"test"}
```

**Comment by mcseemk** [comment]
Try adding the following to the Cargo.toml:

[profile.release]
codegen-units = 1

Then it should start failing. Bizzarre indeed.

**Comment by Licenser** [comment]
Ah yes, that does it, but only for nightly. That's very likely a bug in `nightly` not `simd-json` but it's a good catch, perhaps best to report it to rust. I updated the repo if you want to link it.

**Comment by mcseemk** [comment]
I filed the bug, but the response was discouraging. You can check [there](https://github.com/rust-lang/rust/issues/115040).

**Comment by mcseemk** [comment]
This branch works fine for me. Tried stable/beta/nightly on AMD Ryzen 7950X and Graviton 1/2 - everything OK.

**Comment by Licenser** [comment]
Fix is merged to 0.10.6, I'm closing the ticket, if there are more problems feel free to re-open or make a new one :) thanks for the report.
